api/app/applicationModel/tests.py

```diff
@@ -85,3 +85,168 @@ def test_repo_get_question(self):
+        print('Response Data:', response_data)
+        # Check that the ID is populated
+        self.assertIsNotNone(response_data['id'])
+
+        # Check that the fields are populated correctly in the response
+        self.assertEqual(response_data['event_id'], APPLICATION_FORM_POST_DATA['event_id'])
+        self.assertEqual(response_data['is_open'], APPLICATION_FORM_POST_DATA['is_open'])
+        self.assertEqual(response_data['nominations'], APPLICATION_FORM_POST_DATA['nominations'])
+
+        self.assertEqual(len(response_data['sections']), len(APPLICATION_FORM_POST_DATA['sections']))
+
+        # Check a few fields in the sections themselves
+        section0 = response_data['sections'][0]
+        self.assertIsNotNone(section0['id'])
+        self.assertEqual(section0['name'], APPLICATION_FORM_POST_DATA['sections'][0]['name'])
+        self.assertEqual(len(section0['questions']), len(APPLICATION_FORM_POST_DATA['sections'][0]['questions']))
+
+        section0_question0 = section0['questions'][0]
+        self.assertIsNotNone(section0_question0['id'])
+        self.assertEqual(section0_question0['headline'], APPLICATION_FORM_POST_DATA['sections'][0]['questions'][0]['headline'])
+
+        section1 = response_data['sections'][1]
+        self.assertIsNotNone(section1['id'])
+        self.assertEqual(section1['name'], APPLICATION_FORM_POST_DATA['sections'][1]['name'])
+        self.assertEqual(len(section1['questions']), len(APPLICATION_FORM_POST_DATA['sections'][1]['questions']))
+
+        section1_question0 = section1['questions'][0]
+        self.assertIsNotNone(section1_question0['id'])
+        self.assertEqual(section1_question0['headline'], APPLICATION_FORM_POST_DATA['sections'][1]['questions'][0]['headline'])
+
+        section1_question1 = section1['questions'][1]
+        self.assertIsNotNone(section1_question1['id'])
+        self.assertEqual(section1_question1['headline'], APPLICATION_FORM_POST_DATA['sections'][1]['questions'][1]['headline'])
+
+
+
+class ApplicationFormUpdateTest(ApiTestCase):
+    """
+    Test that an application form's sections and question are updated by an admin with the correct permissions
+    """
+    def _seed_data_update(self):
+
+        self.event = self.add_event()
+        self.system_admin = self.add_user(is_admin=True)
+        self.event_admin = self.add_user(email='user2@user.com')
+        self.event.add_event_role('admin', self.event_admin.id)
+        db.session.commit()
+
+    def test_event_admin_permission(self):
+        """
+        Tests that the application form is being created by a user with the correct permissions
+        """
+        self._seed_data_create()
+
+        response = self.app.post(
+            '/api/v1/application-form',
+            # data=json.dumps(APPLICATION_FORM_POST_DATA),
+            content_type='application/json',
+            headers=self.get_auth_header_for(self.event_admin.email)
+        )
+
+        self.assertEqual(response.status_code, 201)
```

Comment:
A successful PUT request should return status code 200 ("OK"). 201 is for "Created"  :)